code/__HELPERS/maths.dm

```diff
@@ -232,3 +232,24 @@ GLOBAL_LIST_INIT(sqrtTable, list(1, 1, 1, 2, 2, 2, 2, 2, 3, 3, 3, 3, 3, 3, 3, 4,
 	if(round(num) != num)
 		return round(num--)
 	else return num
+
+//proc/get_overlap()
+// Returns a list where [1] is all x values and [2] is all y values that overlap between the given pair of rectangles
+/proc/get_overlap(x1, y1, x2, y2, x3, y3, x4, y4)
+	var/list/region_x1 = list()
+	var/list/region_y1 = list()
+	var/list/region_x2 = list()
+	var/list/region_y2 = list()
+
+	// These loops create loops filled with x/y values that the boundaries inhabit
+	// ex: list(5, 6, 7, 8, 9)
+	for(var/i in (x1 >= x2 ? x2 : x1) to (x1 >= x2 ? x1 : x2))

```

Comment:
Adding intermediate variables would make this more readable

Reply:
could you just use min() and max()?